vcd/resource_vcd_nsxv_firewall_rule.go

```diff
@@ -856,14 +859,42 @@ func orgNetworksIdsToNames(networkIds []string, vdc *govcd.Vdc) ([]string, error
 	for index, networkId := range networkIds {
 		orgVdcNetwork, err := vdc.GetOrgVdcNetworkById(networkId, false)
 		if err != nil {
-			return nil, fmt.Errorf("could not find org network with name %s: %s", networkId, err)
+			return nil, fmt.Errorf("could not find org network with ID %s: %s", networkId, err)
 		}
 		orgNetworkNames[index] = orgVdcNetwork.OrgVDCNetwork.Name
 
 	}
 	return orgNetworkNames, nil
 }
 
+// ipSetIdsToNames looks up IP sets by IDs and returns list of their names
+func ipSetIdsToNames(ipSetIds []string, vdc *govcd.Vdc) ([]string, error) {
+	ipSetNames := make([]string, len(ipSetIds))
+	for index, ipSetId := range ipSetIds {
+		ipSet, err := vdc.GetNsxvIpSetById(ipSetId)
+		if err != nil {
+			return nil, fmt.Errorf("could not find IP set with ID %s: %s", ipSetId, err)
+		}
+		ipSetNames[index] = ipSet.Name
+
+	}
+	return ipSetNames, nil
+}
+
+// ipSetNamesToIds looks up IP set names by their IDs
+func ipSetNamesToIds(ipSetNames []string, vdc *govcd.Vdc) ([]string, error) {
+	ipSetIds := make([]string, len(ipSetNames))
+	for index, ipSetName := range ipSetNames {
+		ipSet, err := vdc.GetNsxvIpSetByName(ipSetName)
```

Comment:
Doesn't api allows to fetch all ipSets? here we can save bunch of requests if locally just browse through and filter out matches

Reply:
Fixed.